torch/csrc/jit/codegen/cuda/scheduler/transpose.cpp

```diff
@@ -432,10 +556,65 @@ void scheduleTranspose(Fusion* fusion, const TransposeParams& params) {
   auto inner_most_id1 = scheduler_utils::innerMostRootDim(reference1);
   auto inner_most_id2 = scheduler_utils::innerMostRootDim(reference2);
 
-  auto inner_most_pos1_in_ref1 =
+  //////////////////////////////////////////
+  // Step 1: Make virtual inner most dims //
+  //////////////////////////////////////////
+
+  // See note [Supporting small transpose dimensions]
+
+  // split big dims so that we have enough dimensions available to merge with
+  // inner-most dims to create the virtual inner-most dim
+  scheduler_utils::splitDims(reference1, params.split_before_tiling);
+  auto merged1 = scheduler_utils::mergeDims(
+      reference1, params.dims_merged_with_1, params.dims_merged_with_2);
+  std::vector<size_t> merged1_vec;
+  if (merged1.has_value()) {
+    merged1_vec.push_back(*merged1);
+  }
+  auto merged2 = scheduler_utils::mergeDims(
+      reference1, params.dims_merged_with_2, merged1_vec);
+  if (merged1.has_value()) {
+    merged1 = merged1_vec[0];
+  }
+
+  // merge with inner most dims to get virtual inner most dims
+  size_t inner_most_pos1_in_ref1 =
       domain_map.getPosMappedTo(reference1, inner_most_id1);
-  auto inner_most_pos2_in_ref1 =
+  size_t inner_most_pos2_in_ref1 =
       domain_map.getPosMappedTo(reference1, inner_most_id2);
```

Comment:
Wouldn't `inner_most_id1` and `inner_most_id` just be the IDs at `merged1` and `merged2`? I thought you just merged `inner_most_id1`  and `inner_most_id2`

Reply:
Wait, you merged everything except the actual `inner_most_id`? Why wouldn't `inner_most_id` be part of that merge in the first place?

Reply:
Because we want to make sure that the `inner_most_id`s are the inner-most dim of the merge, but `scheduler_utils::mergeDims` merges with the order in ref1. For example, if we have
```
T0[I0{2}, I1{2}, I2{1024}, I3{1024}, I4{2}]
with
innermost1 = I4
innermost2 = I0
```
Then we want to merge
```
I1*I2*I0
```
not
```
I0*I1*I2
```
We need to merge `innermost1` separately with some reordering